### ISSUE TITLE
Allow json file upload

### ISSUE THREAD

**Issue by jeff1evesque** [jeff1evesque/machine-learning#1662]
Currently, our file-upload mechanism accepts csv, and xml file uploads.  However, we will accept json file-uploads.  This will require changes within the following files:

- `html_form.js`: need to adjust the `Note:` regarding acceptable file formats
- `convert_upload.py`: need to create the `json_to_dict` method
- `base_data.py`: implement the `json_to_dict` method, in `dataset_to_dict`

**Note:** though, our validation (javascript, and python) validates for the mimetype `text/xml`, we will check if additional validation changes need to be made, for the above suggestions.

**Comment by jeff1evesque** [comment]
During `data_new`, or `data_append` session, `dataset` has the following value when using `sample.json`:

```python
{u'dep-variable-1': [{u'indep-variable-6': 0.002, u'indep-variable-7': 25, u'indep-variable-4': 325, u'indep-variable-5': 54.64, u'indep-variable-2': 98.01, u'indep-variable-3': 0.432, u'indep-variable-1': 23.45}], u'dep-variable-3': [{u'indep-variable-6': 0.002, u'indep-variable-7': 24, u'indep-variable-4': 427, u'indep-variable-5': 75.45, u'indep-variable-2': 101.21, u'indep-variable-3': 0.832, u'indep-variable-1': 22.67}], u'dep-variable-2': [{u'indep-variable-6': 0.001, u'indep-variable-7': 31, u'indep-variable-4': 235, u'indep-variable-5': 64.45, u'indep-variable-2': 92.22, u'indep-variable-3': 0.356, u'indep-variable-1': 24.32}]}
```

**Comment by jeff1evesque** [comment]
We need to correctly define the *feature count* logic within the `json_to_dict` method.

**Comment by jeff1evesque** [comment]
We need to `append`, and not assign observation labels to `self.observation_labels`.

**Comment by jeff1evesque** [comment]
The above remark is encompasses the csv, and xml cases as well, and will be resolved within #1695.
